clothing_spider/clothing_spider/spiders/mohangi_spider.py

```diff
@@ -0,0 +1,150 @@
+"""
+Spider for scraping product information from Mohagni website.
+
+This spider navigates through the Mohagni website to extract details of various products
+including their title, category, URL, images, description, and variants.
+
+Attributes:
+    name (str): The name of the spider.
+    start_urls (list): List of URLs to start crawling from.
+    product_pattern : Regular expression pattern to extract product data from JavaScript.
+    des_pattern : Regular expression pattern to clean JSON data for description.
+    stitched_pattern (re.Pattern): Regular expression pattern to identify stitched product variants.
+
+Methods:
+    parse(self, response): Parses the initial response and initiates category parsing.
+    parse_category(self, response): Extracts category title and pagination links for each category and initiates  parse_products.
+    extract_pagination_links(self, response) : Extracts pagination links from the response and appends '?page=1' to ensure all pages are included. 
+    parse_products(self, response): Parses product pages and initiates product detail parsing for each product url.
+    parse_product_detail(self, response): Parses product detail pages and extracts relevant information.
+    get_varients(self, response): Extracts product variants based on stitching and stitching type.
+
+"""
+
+import json
+import re
+
+import scrapy
+from scrapy import Request
+
+from ..items import ProductItem, SizeItem
+
+
+class MohangiSpider(scrapy.Spider):
+    name = 'mohangi'
+    start_urls = ['https://mohagni.com/']
+
+    product_pattern = re.compile(r'product: ({.*?}),\s*collectionId:')
+    des_pattern = re.compile(r'(?<="gtin14": )([^,\s]+)')
+    stitched_pattern = re.compile(r'\b(STITCHED|S|M|L|XL)\b', re.IGNORECASE)
+    identifier_pattern = r'/products/(\w+-\d+)'
+
+
+    def parse(self, response):
+        links = self.extract_links(response)
+        for link in links:
+            yield response.follow(link, callback=self.parse_category)
+    
+    def parse_category(self, response):
+        category = response.css("h2.collection-hero__title::text")[1].get()
+        pagination_links = self.extract_pagination_links(response)
+        for link in pagination_links:
+            yield response.follow(link, callback=self.parse_products, meta = {"category": category})
+
+    def extract_pagination_links(self, response):
+        # Extract pagination links from response
+        pagination_links = set(response.css("ul.pagination__list a::attr(href)").getall())
+        pagination_links.add(response.url + '?page=1')
```

Comment:
What if the URL already has some parameters? You may use [w3lib](https://w3lib.readthedocs.io/en/latest/) for handling such cases.